sympy/physics/continuum_mechanics/arch.py

```diff
@@ -127,94 +147,482 @@ def apply_load(self,order,label,x1,mag,x2=None,angle=None):
+                eq5 = Eq(moment_A+R_B_y*(self._right_support[0]-self._left_support[0])-\
+                         R_B_x*(self._right_support[1]-self._left_support[1])+\
+                         T*(self._member[2]-self._left_support[1]),0)
+                solution = solve((eq1,eq2,eq3,eq4,eq5),(R_A_x,R_A_y,R_B_x,R_B_y,T))
+
+        elif self._supports['right'] == 'roller':
+            if self._member[2]>=max(self._left_support[1], self._right_support[1]):
+                eq1 = Eq(R_B_x,0)
+                eq2 = Eq(R_A_x+net_x,0)
+                eq3 = Eq(R_A_y+R_B_y+net_y,0)
+                eq4 = Eq(moment_hinge_right+R_B_y*(self._right_support[0]-self._crown_x)+\
+                         T*(self._member[2]-self._crown_y),0)
+                eq5 = Eq(moment_A+R_B_y*(self._right_support[0]-self._left_support[0]),0)
+                solution = solve((eq1,eq2,eq3,eq4,eq5),(R_A_x,R_A_y,R_B_x,R_B_y,T))
+
+            elif self._member[2]>=self._left_support[1]:
+                eq1 = Eq(R_B_x,0)
+                eq2 = Eq(R_A_x+T+net_x,0)
+                eq3 = Eq(R_A_y+R_B_y+net_y,0)
+                eq4 = Eq(moment_hinge_right+R_B_y*(self._right_support[0]-self._crown_x),0)
+                eq5 = Eq(moment_A-T*(self._member[2]-self._left_support[1])+\
+                         R_B_y*(self._right_support[0]-self._left_support[0]),0)
+                solution = solve((eq1,eq2,eq3,eq4,eq5),(R_A_x,R_A_y,R_B_x,R_B_y,T))
+
+            elif self._member[2]>=self._right_support[1]:
+                eq1 = Eq(R_B_x,0)
+                eq2 = Eq(R_A_x-T+net_x,0)
+                eq3 = Eq(R_A_y+R_B_y+net_y,0)
+                eq4 = Eq(moment_hinge_right+R_B_y*(self._right_support[0]-self._crown_x)+\
+                         T*(self._member[2]-self._crown_y),0)
+                eq5 = Eq(moment_A+T*(self._member[2]-self._left_support[1])+\
+                         R_B_y*(self._right_support[0]-self._left_support[0]))
+                solution = solve((eq1,eq2,eq3,eq4,eq5),(R_A_x,R_A_y,R_B_x,R_B_y,T))
+        else:
+            eq1 = Eq(R_A_x + R_B_x + net_x,0)
+            eq2 = Eq(R_A_y + R_B_y + net_y,0)
+            eq3 = Eq(R_B_y*(self._right_support[0]-self._left_support[0])-\
+                     R_B_x*(self._right_support[1]-self._left_support[1])+moment_A,0)
+            eq4 = Eq(moment_hinge_right + R_B_y*(self._right_support[0]-self._crown_x) -\
+                     R_B_x*(self._right_support[1]-self._crown_y),0)
+            solution = solve((eq1,eq2,eq3,eq4),(R_A_x,R_A_y,R_B_x,R_B_y))
+
+        for symb in self._reaction_force:
+            self._reaction_force[symb] = solution[symb]
+        self._bending_moment = - (self._moment_x_func.subs(x,x0) + self._moment_y_func.subs(x,x0) -\
+                                  solution[R_A_y]*(x0-self._left_support[0]) +\
+                                  solution[R_A_x]*(self._shape_eqn.subs({x:x0})-self._left_support[1]))
+
+        angle  = atan(diff(self._shape_eqn,x))
```

Comment:
Running examples 6.2 and 6.3 [here](https://temple.manifoldapp.org/read/a72c8590-5028-4cb4-b3a8-0fb6f9e4026a/section/0ad83309-27c7-4341-aa13-b057374a134c#sec6-1) and the bending moment or axial forces aren't matching.

Reply:
In example 6.3 they are approximating the value to be 11.81 while the accurate value is 11.8080000 which is causing the error , otherwise under under uniformly distributed load, a parabolic arch has a zero bending moment at all positions.

Reply:
```
>>> a = Arch((0,0),(100,0),crown_x=50, crown_y=20)
>>> a.apply_load(0, "Q", 0, -600, 100)
>>> a.solve()
>>> a.bending_moment_at(18)
[0]
```
I am getting 0 for some reason

Reply:
The issue with example 6.2 lies with the limits of the x coordinate , it asks the values just to the left of x=8 but if we put in x = 8 the downward force is also included . at x = 7.9999999 , we get the following results:

![image](https://github.com/user-attachments/assets/de2a9d2f-766c-4de9-8526-f7d75c296aaa)

![image](https://github.com/user-attachments/assets/3615c3c7-6661-4c46-a68d-25bda063f4a9)

these are much closer to the actual results.  I tried using limits to resolve the issue , but since the function we have is continuous for some reason after reaching sufficiently close to x = 8 , i.e. after abs(x-8) < 1e-15 it evaulates it as 8 and not as limit approaching to 8. 

![image](https://github.com/user-attachments/assets/54043ed6-e9cd-485c-a51d-797987881ccb)

![image](https://github.com/user-attachments/assets/fb3e0960-3fb5-46d3-a8af-af1956d0914a)

Reply:
> ```
> >>> a = Arch((0,0),(100,0),crown_x=50, crown_y=20)
> >>> a.apply_load(0, "Q", 0, -600, 100)
> >>> a.solve()
> >>> a.bending_moment_at(18)
> [0]
> ```
> 
> I am getting 0 for some reason

0 should be the correct answer , the example has approximated 11.8080 to 11.81 which has made the answer 75lb.ft.